server/lib/service/dv360.dart

```diff
@@ -27,8 +31,19 @@ class DisplayVideo360Client {
         updateMask: 'entityStatus');
   }
 
-  /// Runs the rule to manipulate DV360 line items.
-  Future<void> run(Rule rule) async {
-    await rule.action.run(this);
+  /// Runs the rule to manipulate DV360 line items and logs the result.
+  Future<void> run(Rule rule, String userId, String ruleId) async {
+    try {
+      await rule.action.run(this);
+    } on ApiRequestError catch (e) {
+      // If there is an API error, return the message returned by the API.
+      return await _firestoreClient.logRunHistory(userId, ruleId, false,
```

Comment:
seems no test coverage for these newly added code? I am ok you add it in later pr.

Reply:
Yes, I'll add it in a later PR.